baseplate/ratelimit/backends/base.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import division
+
+import time
+
+
+class RateLimitBackend(object):
+    """An interface for rate limit backends to implement.
+
+    :param str key: The name of the rate limit bucket to consume from.
+    :param int amount: The amount to consume from the rate limit bucket.
+    :param int allowance: The maximum allowance for the rate limit bucket.
+    :param int interval: The interval to reset the allowance.
+
+    """
+    def consume(self, key, amount, max, bucket_size):
+        raise NotImplementedError
+
+
+def _get_current_bucket(bucket_size):
```

Comment:
It might make sense to to make this a `@staticmethod` of `RateLimitBackend` so subclasses don't have to import this separately.